### ISSUE TITLE
Projects with a non-mod version have no sidedness

### ISSUE THREAD

**Issue by sisby-folk** [modrinth/knossos#1355]
### Describe the bug

After adding a datapack version to any project, the sidedness dropdowns are removed from mod settings, at the mod appears like so:

![image](https://github.com/modrinth/knossos/assets/55819817/a0bf1996-87d1-40d2-b8d5-b971f207b2ad)

This hiding of sidedness should probably only apply when in the "data packs" pane of the site, and the hiding of the dropdowns should probably only occur when _only_ data pack versions exist.

### Steps to reproduce

1. Create a new project
2. Add some mod versions
3. Change the sidedness and have it display as expected
4. add a datapack version of your mod
5. see the sidedness disappear from the mod in all views (not just the datapack tab)
6. see the sidedness options removed from the mod settings page.

### Expected behavior

1. Sidedness selection should remain available for mods with a jar version.
2. Sidedness should be shown in all places _except_ when browsing datapacks.

**Comment by sisby-folk** [comment]
Ah, might as well try to catch two birds - In the profile screen, all project tags also completely disappear, and the project no longer appears in the "mods" pane, _only_ the "datapacks" pane.
![image](https://github.com/modrinth/knossos/assets/55819817/9727c743-3226-42a1-bb87-c6ab8a8a4055)

**Comment by sisby-folk** [comment]
Update: informed that this sidedness disappearing + settings locking also occurs when adding a plugin version to a mod, for example on sidekick
![image-2](https://github.com/modrinth/knossos/assets/55819817/a85039d3-ca7d-4dec-a1ba-51934498b75f)

**Comment by triphora** [comment]
This behavior is intentional. Mods and modpacks are the only project types that have sidedness information; for all other project types, it is entirely irrelevant. Plugins and data packs are always server-side (logical server, anyway), and shaders and resource packs are always client-side.

We're aware that the current system regarding showing/hiding these the side tags is imperfect, as is the moving of the project type from the mods section to the data packs section on user profiles. However, these are issues caused by the more broad hardcoding of the plugin and data pack project types as a subset of mods, which we plan on fixing in API v3. This is being tracked internally as a series of several different issues.

TL;DR: yes it's an issue but it's really caused by another larger, underlying issue that we already plan on fixing but it's a really big thing to have to fix.

**Comment by Minenash** [comment]
Imo, this shouldn't be closed. Some projects have plugins and mods that have client-side components. One of my projects was going to have this, but I never finished the spigot fork
